packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -97,9 +102,11 @@ export function GridOverlays() {
   const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
   const visibleRowCount = useGridSelector(apiRef, gridExpandedRowCountSelector);
   const loading = useGridSelector(apiRef, gridRowsLoadingSelector);
+  const pinnedRowsCount = useGridSelector(apiRef, gridPinnedRowsCountSelector);
 
   const showNoRowsOverlay = !loading && totalRowCount === 0;
   const showNoResultsOverlay = !loading && totalRowCount > 0 && visibleRowCount === 0;
+  const snowNoUnpinnedRowsOverlay = !loading && totalRowCount === 0 && pinnedRowsCount > 0;
```

Comment:
_Optional:_ Do we have customization usecases which require a dedicated slot (`noUnpinnedRowsOverlay`) for this use-case. Could we avoid this slot and use this condition (`pinnedRowCount > 0`) in (`noRowsOverlay`) and render different **localeText** based on it?

Reply:
Idk to be honest, I did it like this because it seems to have templates for the different use cases. This does give higher customizability.

Reply:
+1 to have a single `noRowsOverlay` slot to make the customization easier